File: buddybridge-frontend/src/app/restrict/base/financeiro/pagamento/model/pagamento.ts
import { ContaCaixa } from "../../contacaixa/model/contacaixa";
import { Movimentacao } from "../../movimento/model/movimentacao";

export class Pagamento {
  idPagamento?: number;
  movimentacao?: Movimentacao | null;
  contaCaixa?: ContaCaixa | null;
  dataRecebimento?: Date;
  valorPagamento?: number;

  constructor() {
    this.idPagamento = undefined;
    this.movimentacao = new Movimentacao();
    this.contaCaixa = new ContaCaixa();
    this.dataRecebimento = new Date();
    this.valorPagamento = 0;
  }
}

File: buddybridge-backend/src/main/java/br/com/buddybridge/core/financeiro/pagamento/entity/Pagamento.java
package br.com.buddybridge.core.financeiro.pagamento.entity;

import br.com.buddybridge.core.financeiro.contacaixa.entity.ContaCaixa;
import br.com.buddybridge.core.financeiro.movimentacao.entity.Movimentacao;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "pagamento")
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Pagamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_pagamento")
    private Long idPagamento;

    @ManyToOne
    @JoinColumn(name = "id_movimentacao", nullable = false)
    private Movimentacao movimentacao;

    @ManyToOne
    @JoinColumn(name = "id_conta_caixa", nullable = false)
    private ContaCaixa contaCaixa;

    @Column(name = "data_recebimento", nullable = false)
    private LocalDate dataRecebimento;

    @Column(name = "valor_pagamento", nullable = false)
    private BigDecimal valorPagamento;


}
